test/cram.sh

```diff
@@ -10,6 +11,13 @@ usage() {
 	If the directory is not specified, the current work directory is used
 	instead.
 
+	The options are as follows:
+	  -p/--patch  If there is a difference between the expeted output and the
+	              actual output, the expected output in the file will be patched
+	              by the actual output. Setting the environment variable
+	              '$ENV_CRAM_PATCH' has the same effect. The environment variable
```

Comment:
CRAM_PATCH=1 already does this; I'd just document it as is.

Reply:
Ah I am an idot! I re-invented the wheel in a convoluted manner. Fixed.

Reply:
btw while we are here, I was telling Ken how good your awk skiils are! but for idiots like me, could you please add some comments on the parsing logic so that I can follow?

Reply:
Hey, don't sell yourself short :slightly_smiling_face: My awk skills aren't too impressive, I know awk wizards out there and I'm not yet at that level.

I'll add some comments to explain some of the magic.